src/web/client/utility/UrlBuilder.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { DataverseUrlPathEntityName, HttpMethod, MULTI_ENTITY_URL_KEY, pathParamToSchema, SINGLE_ENTITY_URL_KEY } from "../common/constants";
+import { dataSourcePropertiesMap } from "../common/localStore";
+
+export const getParameterizedRequestUrlTemplate = (entity: string, entityId: string, isSingleEntity: boolean) => {
+    if (isSingleEntity || (entity === DataverseUrlPathEntityName.WEBPAGES && entityId)) {
```

Comment:
why do we have special condition for WEBPAGES?

Reply:
Good catch, will simplify this.